lib/vintage_net_wizard/backend.ex

```diff
@@ -52,9 +52,19 @@ defmodule VintageNetWizard.Backend do
   """
   @callback reset() :: state :: any()
 
+  @doc """
+  Load any previous configurations from the backend
+  """
+  @callback load_configurations() :: [WiFiConfiguration.t()]
+
+  @doc """
+  Check if the backend should be considered configured
+  """
+  @callback configured?([WiFiConfiguration.t()]) :: boolean()
+
   defmodule State do
     @moduledoc false
-    defstruct subscriber: nil, backend: nil, backend_state: nil, configurations: []
+    defstruct subscriber: nil, backend: nil, backend_state: nil, configurations: %{}
```

Comment:
Wouldn't a map lose the configuration order? There's a way to set priorities in the config, but I thought the list was getting mapped directly to the configuration order in the UI.

Reply:
It has always been a map once configurations are loaded, but when we return configurations to the caller we turn it into a list with priority order sorted: https://github.com/nerves-networking/vintage_net_wizard/blob/4fa9eeeb32e30b4eaa87abda07385b59e70285a2/lib/vintage_net_wizard/backend.ex#L347

We use a map internally to simplify updates to a particular SSID. Otherwise, when we update a configuration we would have to always have to check if the current configuration exists and reorder the list after updating. With map we can use `Map.put` and not really care about priority order until the user asks. Also, when we get a priority list there seemed to be a bit of complexity with matching the list of existing configurations with the list of priorities that came through. Again, by using a map it seemed to simply a lot in the implementation in this area.